api/api.py

```diff
@@ -130,14 +130,14 @@ def filter_yaml(yaml: Union[str, List[str], Blob], labels: dict=None, name: str=
 
   Args:
     yaml: Path(s) to YAML, or YAML as a ``Blob``.
-    labels: (optional) return only entities matching these labels. (Matching entities
+    labels: return only entities matching these labels. (Matching entities
       must satisfy all of the specified label constraints, though they may have additional
       labels as well: see the `Kubernetes docs <https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/>`_
       for more info.)
-    name: (optional) the ``metadata.name`` property of entities to match
-    namespace: (optional) the ``metadata.namespace`` property of entities to match
-    kind: (optional) the kind of entities to match (e.g. "service", "deployment", etc.).
-      Case-insensitive. NOTE: doesn't support abbreviations like "svc", "deploy".
+    name: Regexp specifying the ``metadata.name`` property of entities to match
+    namespace: Regexp specifying the ``metadata.namespace`` property of entities to match
+    kind: Regexp specifying the kind of entities to match (e.g. "Service", "Deployment", etc.).
```

Comment:
your change from service -> Service here made me wonder if these regexps should be case-insensitive. there are definitely k8s properties that don't even let you use upper-case to avoid case-sensitivity problems